cmd/tailscale/cli/switch.go

```diff
@@ -178,3 +194,51 @@ func switchProfile(ctx context.Context, args []string) error {
 		}
 	}
 }
+
+func removeProfile(ctx context.Context, args []string) error {
+	if len(args) != 1 {
+		outln("usage: tailscale switch remove NAME")
+		os.Exit(1)
+	}
+	cp, all, err := localClient.ProfileStatus(ctx)
+	if err != nil {
+		errf("Failed to remove account: %v\n", err)
+		os.Exit(1)
+	}
+
+	var profID ipn.ProfileID
+	// Allow matching by ID, Tailnet, or Account
+	// in that order.
+	for _, p := range all {
+		if p.ID == ipn.ProfileID(args[0]) {
+			profID = p.ID
+			break
+		}
+	}
+	if profID == "" {
+		for _, p := range all {
+			if p.NetworkProfile.DomainName == args[0] {
+				profID = p.ID
+				break
+			}
+		}
+	}
+	if profID == "" {
+		for _, p := range all {
+			if p.Name == args[0] {
+				profID = p.ID
+				break
+			}
+		}
+	}
```

Comment:
If you take the search for a match and put it in a separate func, then this same approach will be much cleaner: you can return whenever a match is found rather than checking that there's still no match after each loop completes.

After you make that change, I think this should be ready.